app/src/main/java/com/example/drawingactivity/CustomColorViewModel.kt

```diff
@@ -84,6 +76,43 @@ class CustomColorViewModel : ViewModel() {
         }
     }
 
+    /**
+     * Removes a layer from the list of layers
+     */
+    fun deleteLayer(LayerName: String) {
+        layerList.remove(LayerName)
+    }
+
+    var newName: String? = null
+    var oldName: String? = null
```

Comment:
why is the old name needed

Reply:
I am thinking of a new structure for this where I don't need oldName, I'll look into implementing it shortly